rmw_fastrtps_shared_cpp/src/rmw_take.cpp

```diff
@@ -47,6 +47,10 @@ _assign_message_info(
 {
   message_info->source_timestamp = sinfo->source_timestamp.to_ns();
   message_info->received_timestamp = sinfo->reception_timestamp.to_ns();
+  auto fastdds_sn = sinfo->sample_identity.sequence_number();
+  message_info->publication_sequence_number = (static_cast<uint64_t>(fastdds_sn.high)) <<
+    32 | static_cast<uint64_t>(fastdds_sn.low);
```

Comment:
Did you mean to do this?

```suggestion
  message_info->publication_sequence_number =
    (static_cast<uint64_t>(fastdds_sn.high) << 32) |
    static_cast<uint64_t>(fastdds_sn.low);
```

If not, then maybe there is an extra set of parenthesis?

```suggestion
  message_info->publication_sequence_number = static_cast<uint64_t>(fastdds_sn.high) <<
    32 | static_cast<uint64_t>(fastdds_sn.low);
```